embassy-stm32/src/dcmi.rs

```diff
@@ -0,0 +1,178 @@
+use crate::gpio::sealed::Pin as __GpioPin;
+use crate::gpio::Pin as GpioPin;
+
+#[cfg(feature = "lower-level-peripherals")]
+pub mod ll {
+    pub use super::sealed::*;
+}
+
+mod sealed {
+    use super::*;
+
+    macro_rules! pin {
+        ($name:ident) => {
+            pub trait $name: GpioPin {
```

Comment:
for optional pins (D8+ and the others) you'll want `: OptionalPin` instead of `: GpioPin` because `NoPin` doesn't implement `GpioPin` (and it shouldn't, as it's not a real pin).